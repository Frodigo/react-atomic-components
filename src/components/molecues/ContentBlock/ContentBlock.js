import React from 'react';
import PropTypes from 'prop-types';

import classes from './ContentBlock.module.css';
import {Link} from "../../atoms";

const ContentBlock = (props) => {
    const {
        heading,
        paragraph,
        cta,
        backgroundColor = 'transparent',
        ...rest
    } = props;
    const header = heading ? (
        <h3 className={classes.heading}>{heading}</h3>
    ) : null;
    const content = paragraph ? (
        <p className={classes.paragraph}>{paragraph}</p>
    ) : null;

    const links = props.cta
        ? props.cta.map((block) => (
            <Link key={block.children} href={block.href} variant={block.variant}>{block.children}</Link>
          ))
        : null;

    return (
        <section
            className={classes.block}
            {...rest}
            style={{ backgroundColor }}
        >
            <div className={classes.container}>
                {header}
                {content}

                <div className={classes.links}>{links}</div>
            </div>
        </section>
    );
};

ContentBlock.propTypes = {
    backgroundColor: PropTypes.string,
    cta: PropTypes.arrayOf(
        PropTypes.shape({
            children: PropTypes.string.isRequired,
            disabled: PropTypes.bool,
            href: PropTypes.string.isRequired,
            variant: PropTypes.string
        })
    ),
    heading: PropTypes.string,
    paragraph: PropTypes.string
};

export default ContentBlock;
