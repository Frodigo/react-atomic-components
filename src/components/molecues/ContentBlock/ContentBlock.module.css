.block {
    display: flex;
    flex-direction: column;
    padding: var(--spacing-xxl) var(--spacing-xl);
    text-align: center;
    width: 100%;
}

.heading {
    margin-top: 0;
    margin-bottom: 30px;
}

.paragraph {
    font-size: 16px;
    margin-bottom: 40px;
    line-height: 1.25;
}

@media (min-width: 768px) {
    .container {
        max-width: var(--tablet-container-width);
        margin: 0 auto;
    }

    .links > * {
        margin: 0 var(--spacing-md);
    }

}

@media (min-width: 768px) {
    .block {
        padding-top: 50px;
        padding-bottom: 50px;
    }
}
