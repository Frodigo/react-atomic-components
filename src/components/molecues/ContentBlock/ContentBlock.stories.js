import React from 'react';

import { default as ContentBlockComponent } from './ContentBlock';

export default {
    title: 'Molecues/ContentBlocks',
    component: ContentBlockComponent
};

const Template = (args) => <ContentBlockComponent {...args} />;

export const Primary = Template.bind({});
Primary.args = {
    heading: 'This is heading!',
    paragraph:
        'Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. Curabitur at vestibulum nisi. Nullam vehicula nisi velit. Mauris turpis nisl, molestie ut ipsum et, suscipit vehicula odio. Vestibulum interdum vestibulum felis ac molestie',
    cta: [
        {
            active: true,
            children: 'Get in touch',
            component: 'Link',
            disabled: false,
            href: '/get-in-touch',
            variant: 'primary'
        }
    ]
};
