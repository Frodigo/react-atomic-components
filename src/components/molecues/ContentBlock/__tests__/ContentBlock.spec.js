import React from 'react';
import { render } from '@testing-library/react';

import ContentBlock from '../ContentBlock';

describe('Content block should be rendered', () => {
    const headingMock = 'This is a heading';
    const contentMock = 'Lorem ipsum';
    const ctaMock = [
        {
            href: '#',
            children: 'call to action'
        }
    ];

    test('Content renders correctly', () => {
        const { container } = render(
            <ContentBlock
                heading={headingMock}
                paragraph={contentMock}
                cta={ctaMock}
            />
        );

        expect(container).toMatchSnapshot();
    });
});
