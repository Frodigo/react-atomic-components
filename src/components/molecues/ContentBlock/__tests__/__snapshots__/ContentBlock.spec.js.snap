// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content block should be rendered Content renders correctly 1`] = `
<div>
  <section
    class="block"
    style="background-color: transparent;"
  >
    <div
      class="container"
    >
      <h3
        class="heading"
      >
        This is a heading
      </h3>
      <p
        class="paragraph"
      >
        Lorem ipsum
      </p>
      <div
        class="links"
      >
        <a
          href="#"
        >
          call to action
        </a>
      </div>
    </div>
  </section>
</div>
`;
